pkg/acceptance/barecluster/localcluster.go

```diff
@@ -42,56 +47,83 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/cockroach/pkg/util/stop"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 )
 
-const basePort = 26257
-const dataDir = "cockroach-data"
-
-// CockroachBin is the path to the cockroach binary.
-var CockroachBin = func() string {
-	bin := "./cockroach"
-	if _, err := os.Stat(bin); os.IsNotExist(err) {
-		bin = "cockroach"
-	} else if err != nil {
-		panic(err)
-	}
-	return bin
-}()
+const (
+	listeningURLFile = "cockroachdb-url"
+	pidFile          = "cockroachdb-pid"
+)
 
 // IsUnavailableError returns true iff the error corresponds to a GRPC
 // connection unavailable error.
 func IsUnavailableError(err error) bool {
 	return strings.Contains(err.Error(), "grpc: the connection is unavailable")
 }
 
+// A ClusterConfig holds the configuration for a Cluster.
+type ClusterConfig struct {
+	Ephemeral   bool
+	Binary      string
+	AllNodeArgs []string
+	NumNodes    int
+	DataDir     string
+	PerNodeCfg  map[int]NodeCfg
+
+	DB         string
+	NumWorkers int
+}
+
+// NodeCfg . RENAME Config
+type NodeCfg struct {
+	DataDir             string
+	Addr                string
+	ExtraArgs, ExtraEnv []string
+	RPCPort, HTTPPort   int // zero for auto-assign
```

Comment:
For tests, do we ever want these to be non-zero? Ah, I see you're using these for backwards compatibility with `zerosum` and `allocsim`. Carry on.